### PR TITLE
Wallet service fixes

### PR DESCRIPTION
**Pull request by yontyon**
* **Problem Overview**  
  This PR fixes the following:
1. It saves the coins in a map instead of a list, such that a coin cannot be stored twice
2. Add functionality to get the coins identifier & values from the wallet service
* **Testing Done**  
  uinttests

### CODE REVIEW

File: utt/utt-client-api/src/User.cpp [hunk 1]
```diff
@@ -583,7 +586,8 @@ void User::debugOutput() const {
     std::cout << '\n';
   };
 
-  for (const auto& coin : pImpl_->coins_) {
+  for (const auto& [_, coin] : pImpl_->coins_) {
+    (void)_;
```
Comment (eyalrund) [inline utt/utt-client-api/src/User.cpp:590]: how did it compile before without a warning?

File: utt/privacy-wallet-service/proto/api/v1/wallet-api.proto [hunk 2]
```diff
@@ -98,6 +98,7 @@ message GetStateResponse {
     uint64 budget = 1[jstype = JS_STRING];
     uint64 balance = 2[jstype = JS_STRING];
     string user_id = 3;
+    map<string, uint64> coins = 4;
```
Comment (bsenthilr) [inline utt/privacy-wallet-service/proto/api/v1/wallet-api.proto:101]: how do you attribute jstype = JS_STRING for this map? it is required for JS to work

**Comment by bsenthilr** [comment]
The changes LGTM. The annotation for JS can be taken care on next PR.

**Review by bsenthilr — APPROVED** [review]

**Comment by teoparvanov** [comment]
The fixes look good, thanks @yontyon !

**Review by teoparvanov — APPROVED** [review]

File: utt/utt-client-api/src/User.cpp (hunk 1 above)
Comment (yontyon) [reply utt/utt-client-api/src/User.cpp:590]: previously, coins was a vector

**Pull request merged**